app/src/main/java/com/github/gogetters/letsgo/GoView.kt

```diff
@@ -0,0 +1,153 @@
+package com.github.gogetters.letsgo
+
+import android.content.Context
+import android.graphics.*
+import android.util.Log
+import android.view.MotionEvent
+import android.view.View
+import androidx.core.content.ContextCompat
+import com.github.gogetters.letsgo.game.Board
+import com.github.gogetters.letsgo.game.BoardState
+import com.github.gogetters.letsgo.game.Stone
+import com.github.gogetters.letsgo.game.Point
+import kotlin.math.roundToInt
+
+
+class GoView(context: Context, private val boardSize: Board.Size) : View(context) {
+
+    constructor(context: Context): this(context, Board.Size.withSize(9))
+
+
+    private val boardImageID = when (boardSize) {
+        Board.Size.SMALL -> R.drawable.board_9
+        Board.Size.MEDIUM -> R.drawable.board_13
+        Board.Size.LARGE -> R.drawable.board_19
+    }
+
+    lateinit var inputDelegate: InputDelegate
+
+    private val boardImage = ContextCompat.getDrawable(context, boardImageID)
+    private val whiteStoneImage = ContextCompat.getDrawable(context, R.drawable.white)
+    private val blackStoneImage = ContextCompat.getDrawable(context, R.drawable.black)
+
+    private var marginRatio = 120F / 880F
+
+
+    private var board = BoardState.emptyBoard(boardSize)
+
+
+    private fun coordToPoint(x: Float, y: Float): Point {
+
+        val marginX: Float = width.toFloat() * marginRatio
+        val marginY: Float = height.toFloat() * marginRatio
+
+        val maxX = width.toFloat() - 2F * marginX
+        val maxY = height.toFloat() - 2F * marginY
+
+        val squareWidth = maxX / (boardSize.size - 1)
+        val squareHeight = maxY / (boardSize.size - 1)
+
+        width.toFloat() * marginRatio
```

Comment:
What is this supposed to do?

Reply:
Does nothing... not sure why my linter didn't pick it up. I will remove it